### DISCUSSION TITLE
What's the zustand way to init on load?

### DISCUSSION THREAD

**Discussion by the-artifabrian** [pmndrs/zustand#1560] [Q&A]
So I need asset/ticker data for some components, while on others, I don't.
Initially, I was thinking of calling the fetch function in the store file like in the example below, but this would fetch the data on all routes:
```
export const useAssetsStore = create()((set) => {
  return {
    assets: [],
    fetchAssets: async () => {
       // make api call here
       // then set({ assets })
    }
  }
})

const { assets, fetchAssets } = useAssetsStore().get()

if (!assets.length) {
  fetchAssets()
}
```

But then I wondered if this would also be acceptable, the store should get initialized only when the hook is used in a component:
```
export const useAssetsStore = create()((set, get) => {
  const assets = get().assets

  if (!assets.length) {
    // make api call here
    // then set({ assets })
  }

  return {
    assets: [],
  }
})
```

Thoughts?

**Comment by dai-shi** [comment]
Unfortunately, the second one doesn't work. The function we pass to `create()` is an initializer, which will be invoked immediately just once.
The first one seems good. You just need to call `fetchAssets()` in a way as you want.
